### PR TITLE
feat: expose gcf memory param in `remote_function`

### PR DESCRIPTION
**Pull request by shobsi**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes internal issue 346857621 🦕

### CODE REVIEW

File: tests/system/large/test_remote_function.py [hunk 1]
```diff
@@ -1800,3 +1800,63 @@ def float_parser(row):
         cleanup_remote_function_assets(
             session.bqclient, session.cloudfunctionsclient, float_parser_remote
         )
+
+
+@pytest.mark.parametrize(
+    ("memory_mib_args", "expected_memory"),
+    [
+        pytest.param({}, "1024Mi", id="no-set"),
+        pytest.param({"cloud_function_memory_mib": None}, "256M", id="set-None"),
```
Comment (TrevorBergeron) [inline tests/system/large/test_remote_function.py:1809]: Why different values for unset vs None?

File: tests/system/large/test_remote_function.py [hunk 2]
```diff
@@ -1800,3 +1800,63 @@ def float_parser(row):
...
+        pytest.param({"cloud_function_memory_mib": 128}, "128Mi", id="set-128"),
+        pytest.param({"cloud_function_memory_mib": 1024}, "1024Mi", id="set-1024"),
+        pytest.param({"cloud_function_memory_mib": 4096}, "4096Mi", id="set-4096"),
+        pytest.param({"cloud_function_memory_mib": 32768}, "32768Mi", id="set-32768"),
```
Comment (TrevorBergeron) [inline tests/system/large/test_remote_function.py:1813]: What happens if users provides a value that is within min-max bounds but is not a power of two?

File: tests/system/large/test_remote_function.py (hunk 1 above)
Comment (shobsi) [reply tests/system/large/test_remote_function.py:1809]: unset should be backward compatible, meaning we should continue to set bigframes default of 1024M. `None` is a way to indicate that any explicit memory from the bigframes side should not be set, as added in the documentation "If you want to let the default memory of cloud functions be allocated, pass `None`". There is another param `cloud_function_timeout` with similar behavior. I acknowledge the confusion though, how about I send a separate PR to make unset and None behavior same in both param for consistency?

File: tests/system/large/test_remote_function.py (hunk 2 above)
Comment (shobsi) [reply tests/system/large/test_remote_function.py:1813]: there is some server side logic to mod/ceil it the nearest well known number.

**Review by TrevorBergeron — APPROVED** [review]

**Pull request merged**